contracts/modules/staking/StakingProducts.sol

```diff
@@ -162,12 +162,32 @@ contract StakingProducts is IStakingProducts, MasterAwareV2, Multicall {
       }
 
       if (_param.setTargetPrice) {
+
         if (_param.targetPrice > TARGET_PRICE_DENOMINATOR) {
           revert TargetPriceTooHigh();
         }
+
         if (_param.targetPrice < globalMinPriceRatio) {
           revert TargetPriceBelowMin();
         }
+
+        // if this is an existing product, when the target price is updated we need to calculate the
+        // current base price using the old target price and update the bumped price to that value
+        // uses the same logic as calculatePremium()
+        if (_product.bumpedPriceUpdateTime != 0) {
```

Comment:
Note: I think we need to make a var `isNewProduct` since `product.bumpedPriceUpdateTime` is already set to timestamp above.